src/account/components/OptionalAuth.js

```diff
@@ -37,6 +39,12 @@ const OptionalAuth = ({ children }) => {
     )
   );
 
+  useEffect(() => {
+    if (fetching === false && !user && hasToken) {
+      dispatch(signOut());
```

Comment:
question: in `RequireAuth`, we dispatch `setHasToken(false)` instead of `signOut()` in a similar condition, why is it different?